frontend/SettingsCheckbox.js

```diff
@@ -45,8 +45,8 @@ class SettingsCheckbox extends React.Component<Props, State> {
   }
 
   componentDidMount(): void {
-    if (!this.props.state !== this._defaultState) {
-      this.props.onChange(this._defaultState);
+    if (this.props.state && (this.props.state !== this._defaultState)) {
+      this.props.onChange(this.props.state);
```

Comment:
This feels weird to me. If both `onChange` and `state` are coming from `props` (aka being passed in) then it doesn't seem right that our React component would need to tell the caller that props have "changed".

I'm going to think about this a little bit.